Sync using Netflix api

MrMamen opened the pull request:
Using Netflix API for activity log. I expect this bug to fix issue #13 #19 and probably #15

tegon commented:
This looks good, thanks. I just some considerations before merge it:
- The `content-controller.js` and `search.js` are from your other pull request right? If so, could you check out them to master? This is could cause some conflicts when we merge.
- This is a breaking change, so the test suite is broken. Could you update the tests?

Thanks again.

MrMamen commented:
I should obviously have started from a clean master. Since it will take some time to understand the testing framework I will rather wait until the other PR has been merged, then fix this branch rather than rebasing it on another master right now.

MrMamen commented:
@tegon I've rebased this so it does not include anything from the other Pull Request ( #18 ). I have also updated all the tests. I see that there is one test that fails on your server, but does not fail locally. I can't really understand why it is like this, but I hope this doesn't stop you from merging the PR.

tegon commented:
@MrMamen Ok, thanks. I'll see this as soon as I can.

tegon commented on tests/content/viewing-activity-parser-test.js:
```diff
@@ -8,36 +9,36 @@ describe('ViewingActivityParser', function() {
     it('creates the activity when synced_at equals to item date', function() {
-      var activity = ViewingActivityParser.parse(new Date('10/2/2015'), window.episodeActivity);
+      var activity = ViewingActivityParser.parse(moment('10/2/2015', 'M/D/YYYY'), window.episodeActivity);
       expect(activity instanceof ViewingActivity).toBeTruthy();
```
@MrMamen Just ran this test here, it is failing because the date passed is before the `syncedAt`. This [condition](https://github.com/tegon/traktflix/pull/20/files#diff-fd41d2ff17a8b49f44d8e8c26de6084bR12) is returning `false` so the activity will be `undefined` failing the expect in the next line. If you change the first argument here to `moment('10/1/2015', 'M/D/YYYY')` it will pass.

MrMamen replied:
Ahh thanks. I belive this is another timezone issue. I've put the time in absolute time in netflix-viewing-activity, while this might both be before and after the original date depending on location. Thanks.
